nav2_navfn_planner/src/navfn_planner.cpp

```diff
@@ -346,6 +346,10 @@ NavfnPlanner::makePlan(
         "Failed to create a plan from potential when a legal"
         " potential was found. This shouldn't happen.");
     }
+  } else {
+    throw nav2_core::GoalOccupied(
```

Comment:
This shouldn't be necessary?

`return !plan.poses.empty();` causes NoPathCanBeFound exception thrown https://github.com/ros-planning/navigation2/blob/933f4a9cfcae3ebd2860e057172910b8d23360c5/nav2_navfn_planner/src/navfn_planner.cpp#L193-L194 since none of the path is populated.